File: modular-framework/modules/llm-tester/src/sse.js
import { logDebug } from './logger.js';

export function openSSE(res) {
  res.setHeader("Content-Type", "text/event-stream");
  res.setHeader("Cache-Control", "no-cache");
  res.setHeader("Connection", "keep-alive");
  res.flushHeaders?.();
  try { logDebug('LT SSE open'); } catch {}

}
export function send(res, obj) {
  try { logDebug('LT SSE send', { type: obj?.type }); } catch {}

  res.write(`data: ${JSON.stringify(obj)}\n\n`);
}
export function close(res) {
  try { logDebug('LT SSE close'); } catch {}

  res.write("event: done\ndata: {}\n\n");
  res.end();
}

File: tempchat/branchy-chat-backend/app/utils/sse.py
from fastapi import Response
from typing import AsyncGenerator

SSE_HEADERS = {"Content-Type": "text/event-stream", "Cache-Control": "no-cache"}

def sse_event(event: str | None = None, data: str = "") -> str:
    out = []
    if event:
        out.append(f"event: {event}")
    for line in data.splitlines() or [""]:
        out.append(f"data: {line}")
    out.append("")
    return "\n".join(out)

class SSE(Response):
    media_type = "text/event-stream"